detekt-tooling/src/main/kotlin/io/github/detekt/tooling/api/spec/ProjectSpec.kt

```diff
@@ -16,4 +17,11 @@ interface ProjectSpec {
      * Paths to analyze. Works with files and directories.
      */
     val inputPaths: Collection<Path>
+
+    /**
+     * The analysis mode used by detekt. 'light' mode means detekt will analyze code in individual files but cannot use
+     * compiler information like types, symbols and smart casts. 'full' mode allows rules to use additional information
+     * from the compiler when analyzing code.
```

Comment:
```suggestion
     * The analysis mode used by detekt.
     *  * 'light' mode means detekt will analyze code in individual files
     *    but cannot use compiler information like types, symbols and smart casts.
     *  * 'full' mode allows rules to use additional information from the compiler when analyzing code.
```
also, maybe these bullet points are redundant? and should be instead documented in 1 place, on the enum!?

Reply:
Yeah we could move it - the KDoc here would say "analysis mode used by detekt" then refer to the enum, and these descriptions go on each enum option. I think that makes sense.

I won't change it yet so others can weigh in.

Reply:
Yep that's what I meant.